Update Extension for Libreoffice 6.3/6.4

kalinjul opened the pull request:
Update Extension for Libreoffice 6.4

This PR makes the extension run on LibreOffice 6.4.
It does not change the actual checking code.
The extension shows up as additional sidebar
while the menubar button (still) doesn't do anything.
- fix compile errors with newer API
- fix sidebar visibility
- define sidebarpanel in Sidebar.xcu instead of WriterWindowState.xcu
- add icon for sidebar

bertfrees commented on accessodf-addon/src/main/java/be/docarch/accessodf/ooo/toolpanel/AccessibilityPanel.java:
```diff
@@ -1421,4 +1425,17 @@ private static String unoURLtoURL(String unoURL) {
+	@Override
+	public LayoutSize getHeightForWidth (final int nWidth)
+	{
+//		final int nMinHeight = 100;
+//		final int nMaxHeight = Math.max(nWidth, nMinHeight);
```
Can I remove these comments?

bertfrees commented on accessodf-addon/src/main/oxt/registry/data/org/openoffice/Office/UI/Factories.xcu:
```diff
@@ -6,13 +6,14 @@
         </prop>
         <prop oor:name="Name">
-          <value>be.docarch.accessodf.ooo.toolpanel</value>
+          <!-- <value>be.docarch.accessodf.ooo.toolpanel</value> -->
```
Why is this commented?

bertfrees commented on accessodf-addon/src/main/oxt/registry/data/org/openoffice/Office/UI/Sidebar.xcu:
```diff
@@ -0,0 +1,57 @@
+
+    <node oor:name="PanelList">
+      <node oor:name="AccessibilityPanel" oor:op="replace">
+        <!-- <prop oor:name="Title" oor:type="xs:string">
+          <value xml:lang="en-US">Accessibility</value>
+          <value xml:lang="de-DE">Barrierefreiheit</value>
+        </prop> -->
```
Again, why the comment?

bertfrees commented:
Hi! Thank you very much for the PR. I quickly skimmed through the changes and made a few minor remarks about comments in the code. The change itself looks good. I haven't tested it yet. I might do that when I find some time, but maybe I will just merge it because I trust you have tested it.

kalinjul commented:
Thanks for your comments.
I removed the commented lines, as propably suspected those where leftovers.
I tested it so far that the extension is showing, does it's checking and shows warnings etc., however, I didn't verify the checks themselves work correctly. So far it seems plausible.

It seems there is also work going on to include accessibility checks in LibreOffice [1], which may render this extension useless, but nevertheless better to have it working with recent versions than not, i guess.

[1] https://git.libreoffice.org/core/+log/8a78de65b5472b933267601cd484f3ff5e75c2a3/svx/source/dialog/AccessibilityCheckDialog.cxx

bertfrees commented:
@kalinjul Thanks for the additions. Some questions:

> Show error dialog if a check failed

Can you give an example of an exception that happened and resulted in "congratulations", or is this only theoretical?

> Fix graphic traversal

Can you explain what exactly happened in this commit?

> Fix detection of forms

Can you explain what exactly happened in this commit?

kalinjul commented:
FYI, i went through translating the error messages and tried to reproduce the errors/warnings in my document. When i found checks are not working, I tried to fix them, so the extension should now work quite well in LO 6.3+.

> 
> 
> @kalinjul Thanks for the additions. Some questions:
> 
> > Show error dialog if a check failed
> 
> Can you give an example of an exception that happened and resulted in "congratulations", or is this only theoretical?

This happened when you included any image in your document - see next commit.
While i fixed this particular problem, others may occur, so i figured it would be better to show an error than to silently ignore everything and report success.

> 
> > Fix graphic traversal
> 
> Can you explain what exactly happened in this commit?

The graphic object doesn't have a "GraphicURL" field anymore, so it throw an exception.
To be specific, the field is still there, but access is not allowed.
See https://tomazvajngerl.blogspot.com/2018/03/improving-image-handling-in-libreoffice.html

> 
> > Fix detection of forms
> 
> Can you explain what exactly happened in this commit?

I had a sample document which reported xSuppForms.hasForms() = true, but didn't hat a form and xSuppForms.getForms().getElementNames().length was 0. So i added this line to the check.
